code/modules/research/xenoarchaeology/machinery/analysis_base.dm

```diff
@@ -51,67 +49,66 @@
 	..()
 	if (stat & NOPOWER && scan_process)
 		stop()
-
 	else
 		update_icon()
 
 /obj/machinery/anomaly/process()
-	//not sure if everything needs to heat up, or just the GLPC
-	var/datum/gas_mixture/env = loc.return_air()
-	var/environmental_temp = env.temperature
+	//First we deal with the machine's task
 	if(scan_process)
-		// Shouldn't be reachable, still can't hurt.
-		if(stat & NOPOWER)
+		if (stat & (NOPOWER|BROKEN))
```

Comment:
You only check if it's broken or has no power if you're scanning. I don't see any reason for that, it should also check it if it's idle.

But also it doesn't return. So if it's broken or has no power then it processes the temperature changes as if it was idle/working

Reply:
That specifically is intended. In the future I'm thinking of having temperature management be a machine flag so we can have the same work with tcomms machines for example. But yes, I want the temperature to change independantly of whether it's broken or powered (and whether it's broken or powered only matters if it was scanning, in which case it'll need to stop)

But now that you mention it, I'll make sure it doesn't produce any extra heat while broken/unpowered but I still want the entropy to regulate the temperature between it and the environment.